frontend/src/components/Modal.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect } from 'react';
+import ReactDOM from 'react-dom';
+import { useModalStack } from '@/context/ModalContext';
+import { useModal } from '@/hooks/useModal';
+
+type ModalProps = {
+  children: React.ReactNode;
+  id: string;
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+export const Modal: React.FC<ModalProps> = ({
+  id,
+  isOpen,
+  onClose,
+  children,
+}) => {
+  // eslint-disable-next-line react/hook-use-state
+  const modalRef = useModal(isOpen, onClose);
+  const { modalStack, pushModal, popModal } = useModalStack();
+
+  useEffect(() => {
+    // eslint-disable-next-line no-debugger
+    if (isOpen) {
+      pushModal(id);
+    } else {
+      popModal();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isOpen, id]);
+
+  if (!isOpen) return null;
+
+  return ReactDOM.createPortal(
+    <div
+      aria-modal="true"
+      className="fixed inset-0 bg-neutral-800 bg-opacity-50 "
+      ref={modalRef}
+      role="dialog"
```

Comment:
use this role here for more semantic represenation and accesibility, the inner div with presentation role was added given the linting rules.